common/src/autogluon/common/utils/utils.py

```diff
@@ -62,7 +62,6 @@ def setup_outputdir(path, warn_if_exist=True, create_dir=True, path_suffix=None)
                 f'Warning: path already exists! This predictor may overwrite an existing predictor! path="{path}"'
             )
     path = os.path.expanduser(path)  # replace ~ with absolute path if it exists
-    path = os.path.abspath(path)
```

Comment:
Maybe we can wrap this block into
```python
if not path.startswith("s3://"):
    path = os.path.expanduser(path)  # replace ~ with absolute path if it exists
    path = os.path.abspath(path)
```
I believe there are some edge cases against which the current `path = os.path.abspath(path)` command protects (though it does introduce another bug with loading from S3).

Reply:
Yes, this looks like the best solution.

Reply:
What happens if one calls `.expanduser()` on an S3 path? I think likely nothing, as S3 paths do not contain/start with `~`. 

Otherwise, the current suggested if-case would change this behavior, too.

Reply:
In this case I think what @shchur proposed should accommodate S3 paths. I will update the PR.